front50-s3/src/main/java/com/netflix/spinnaker/front50/model/S3StorageService.java

```diff
@@ -50,22 +40,35 @@
   private final String bucket;
   private final String rootFolder;
   private final Boolean readOnlyMode;
+  private final String region;
 
   public S3StorageService(ObjectMapper objectMapper,
                           AmazonS3 amazonS3,
                           String bucket,
                           String rootFolder,
-                          Boolean readOnlyMode) {
+                          Boolean readOnlyMode,
+                          String region) {
     this.objectMapper = objectMapper;
     this.amazonS3 = amazonS3;
     this.bucket = bucket;
     this.rootFolder = rootFolder;
     this.readOnlyMode = readOnlyMode;
+    this.region = region;
   }
 
   @Override
   public void ensureBucketExists() {
-    // do nothing
+    List<Bucket> buckets = amazonS3.listBuckets();
```

Comment:
[headBucket](http://docs.aws.amazon.com/AWSJavaSDK/latest/javadoc/com/amazonaws/services/s3/AmazonS3Client.html#headBucket-com.amazonaws.services.s3.model.HeadBucketRequest-) is probably a better fit here than iterating all existing buckets.